curve_apps/driver.py

```diff
@@ -38,42 +39,59 @@ class BaseCurveDriver(BaseDriver):
     _default_name: str
 
     def __init__(self, parameters: BaseData | InputFile):
+        self._out_group = None
         if isinstance(parameters, InputFile):
             parameters = self._parameter_class.build(parameters)
 
         # TODO need to re-type params in base class
         super().__init__(parameters)
 
-    def run(self):
-        """
-        Run method of the driver.
+    @property
+    def out_group(self) -> ContainerGroup | None:
+        """Output container group."""
+
+        if self._out_group is None and self.params.output.out_group is not None:
+            self._out_group = ContainerGroup.create(
```

Comment:
We could start using UIJsonGroup here

Reply:
If you do, then you can use this to append the json to options

`options = InputFile.stringify(
                InputFile.demote(self.params.input_file.ui_json)
            )`